File: 00_ETC/2490.cpp
#include <iostream>
#include <vector>
using namespace std;

int main() {
    ios_base::sync_with_stdio(false);
    cin.tie(NULL);
    cout.tie(NULL);

    for (int i = 0; i < 3; ++i) {
        vector<int> v(4);
        for (int j = 0; j < 4; ++j) {
            cin >> v[j];
        }
        int cnt0 = 0, cnt1 = 0;
        for (int j = 0; j < 4; ++j) {
            if (v[j] == 0) {
                ++cnt0;
            } else if (v[j] == 1) {
                ++cnt1;
            }
        }

        if (cnt0 == 1 && cnt1 == 3) {
            cout << "A" << endl;
        } else if (cnt0 == 2 && cnt1 == 2) {
            cout << "B" << endl;
        } else if (cnt0 == 3 && cnt1 == 1) {
            cout << "C" << endl;
        } else if (cnt0 == 4) {
            cout << "D" << endl;
        } else if (cnt1 == 4) {
            cout << "E" << endl;
        }
    }
    return 0;
}

//
// Created by 최승우 on 25. 8. 10.
//

File: 00_ETC_PY/2490.py
for i in range(3):
    x = list(map(int, input().split()))
    cnt0 = cnt1 = 0
    for v in x:
        if v == 0:
            cnt0 += 1
        elif v == 1:
            cnt1 += 1
    if cnt0 == 1 and cnt1 == 3:
        print("A")
    elif cnt0 == 2 and cnt1 == 2:
        print("B")
    elif cnt0 == 3 and cnt1 == 1:
        print("C")
    elif cnt0 == 4:
        print("D")
    elif cnt1 == 4:
        print("E")
